src/olympia/versions/models.py

```diff
@@ -825,6 +826,20 @@ class Meta:
     def get_application_display(self):
         return six.text_type(amo.APPS_ALL[self.application].pretty)
 
+    def get_latest_application_version(self):
+        def _fetch_version():
+            qs = (
+                AppVersion.objects
+                .filter(
+                    ~models.Q(version__contains='*'),
+                    application=self.application)
+                .order_by('-version_int'))
+            return qs.first()
+
+        cache_key = 'appversion:{}:latest'.format(self.application)
+        return cache_get_or_set(
+            cache_key, _fetch_version, timeout=60 * 60 * 24)
```

Comment:
Actually… thinking about this after a weekend… I don't think it has to be cached since it's used only very rarely. I'll keep it for the review though for now, removing it is easy. Opinions about that?

Reply:
I think you can remove the caching right now. The query should be pretty fast already, there are not a lot of entries, there are indexes helping it already, and as you pointed out, it's going to be used so rarely it probably doesn't matter.

Reply:
Done in 6db8a494cd